pkg/otlp/utils.go

```diff
@@ -0,0 +1,82 @@
+package otlp
+
+import (
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+
+	"github.com/siglens/siglens/pkg/utils"
+	log "github.com/sirupsen/logrus"
+	"github.com/valyala/fasthttp"
+	"google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+func getDataToUnmarshal(ctx *fasthttp.RequestCtx) ([]byte, error) {
+	// We don't support JSON requests yet.
+	if contentType := utils.GetContentType(ctx); contentType != utils.ContentProtobuf {
+		log.Errorf("getDataToUnmarshal: got a non-protobuf request. Got Content-Type: %s", contentType)
+		return nil, fmt.Errorf("Expected a protobuf request")
```

Comment:
we should probably do:
```
if ... not protobuf {
        return nil, fmt.Errorf(getDataToUnmarshal: got a non-protobuf request. Got Content-Type: %s", contentType)
}

Reply:
and have the caller log the error msg, so that 
1) we get only one log line
2) when we implement logs and metrics, we can differentiate easily at the caller for which type this error was